src/core/model/Stroke.cpp

```diff
@@ -379,6 +386,24 @@ auto Stroke::getAvgPressure() const -> double {
            this->points.size();
 }
 
+void Stroke::updateBoundsLastTwoPressures(double fallbackPressure) {
+    if (!sizeCalculated || this->points.empty()) {
+        return;
+    }
+
+    Point& p = this->points.back();
+    auto const pointCount = this->getPointCount();
+    if (pointCount >= 2) {
+        Point& p2 = this->points[pointCount - 2];
+        double pressure = p2.z;
+
+        updateBounds(Element::x, Element::y, Element::width, Element::height, snappedBounds, p, 0.5 * pressure);
+        updateBounds(Element::x, Element::y, Element::width, Element::height, snappedBounds, p2, 0.5 * pressure);
+    } else if (pointCount == 1) {
```

Comment:
This case should be totally excluded from `updateBoundsLastTwoPressures` (like with an assert).